src/index.ts

```diff
@@ -1,32 +1,51 @@
-import { Board, renderBoard, updateBoardState } from "./board/board";
+import {
+  Board,
+  highlightAllowedTiles,
+  renderBoard,
+  updateBoardState,
+} from "./board/board";
 import { initCLI } from "./cli";
 import { loadGameConfigFromFile } from "./gameLoader";
 import { prompt } from "./prompt/prompt";
 import { switchPlayer } from "./player/player";
 import { checkUserMove } from "./move/move";
 import { initGameState } from "./game/state";
+import { renderTurnDisplay } from "./turn";
 
 initCLI();
 const gameConfig: Board = loadGameConfigFromFile();
 
-let gameState = initGameState();
-
-renderBoard(gameConfig);
+let currentGameState = initGameState();
+const highlightedInitialBoard = highlightAllowedTiles(
+  gameConfig,
+  currentGameState
+);
+renderTurnDisplay(currentGameState.turnNumber);
+renderBoard(highlightedInitialBoard);
 
 (async () => {
-  while (gameState.isRunning) {
-    const action = await prompt(gameState, gameConfig);
-    const userCanMove = checkUserMove(gameConfig, action, gameState);
-    gameState = userCanMove.gameState;
-    if (!userCanMove.allowedMove) continue;
+  while (currentGameState.isRunning) {
+    const action = await prompt(currentGameState, highlightedInitialBoard);
+    const userCanMove = checkUserMove(gameConfig, action, currentGameState);
+
+    const { gameState, allowedMove } = checkUserMove(
+      gameConfig,
+      action,
+      currentGameState
+    );
```

Comment:
Duplicate (and probably unnecessary) call to `checkUserMove`

Reply:
It's gone.